src/cmd/generate-v1/main.go

```diff
@@ -27,6 +27,7 @@ func main() {
 
 		ModuleDirectory:   *moduleDataDir,
 		ProviderDirectory: *providerDataDir,
+		Logger:            slog.Default(),
```

Comment:
What's the reason for having the logger as part of the owning type?

I feel like there are cases where we are immediately going to create a new logger when we call `logger.With` at the top level exported methods of this package.